client/kubernetes/pom.xml

```diff
@@ -0,0 +1,86 @@
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+  xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+  <modelVersion>4.0.0</modelVersion>
+
+  <artifactId>jolokia-kubernetes</artifactId>
+  <packaging>bundle</packaging>
+  <parent>
+    <groupId>org.jolokia</groupId>
+    <artifactId>jolokia-client-parent</artifactId>
+    <version>1.6.3-SNAPSHOT</version>
+    <relativePath>../pom.xml</relativePath>
+  </parent>
+
+  <name>jolokia-kubernetes</name>
+
+  <dependencies>
+
+
+    <dependency>
+      <groupId>org.jolokia</groupId>
+      <artifactId>jolokia-jmx-adapter</artifactId>
+      <version>${project.version}</version>
+      <!-- as jolokia-jmx-adapter has its own bundle, keep them separate -->
+      <scope>provided</scope>
+    </dependency>
+    <dependency>
+      <groupId>io.kubernetes</groupId>
+      <artifactId>client-java</artifactId>
```

Comment:
Does the fabric8io require some fabric8 components in the cluster, or does it work on any kubernetes cluster? I chose kubernets project own Java client as a default. It also supports authentication towards Azure AKS. The need to bridge http components is naturally a downside. The component itself is very thin so it could be quite easy to create more components such as fabric8:  or openshift: to optimize use of specific functionality.

Reply:
[fabric8io/kubernetes-client](https://github.com/fabric8io/kubernetes-client) works on any cluster ist just a client library that talks over REST to the Kubernetes API servers. It has a much nicer API IMO (builder pattern vs methods with 10+ args) and is also much more popular than the "official" Java client. Whether it supports Azure authentication, I'm not sure but I'm also not sure what Azure does different than any other cloud vendor (as this works with "standard" kube config based authentication).

Reply:
Plan:
1. Sort out the proposed improvements.
2. Experiment with fabric8 , either as a sibling module or a new branch on top of this one.

Reply:
Pushed changes for 1.